docs/intro-to-dune/index.md

```diff
@@ -0,0 +1,284 @@
+npm script. In the world of OCaml, usage of [the Make build automation
+tool](https://en.wikipedia.org/wiki/Make_(software)) is pretty much ubiquitous,
+and this is not likely to change. So let's take a short detour to explain the
+`Makefile` we just added.
+
+## Anatomy of Makefile
+
+In case you're not familiar with Make, here's a breakdown of
+`src/counter/Makefile`:
+
+<<< @/../src/counter/Makefile{make}
+
+- There's only a single rule here, and its *target* is `serve`.
+- `.PHONY: serve` is used to indicate that `serve` is a [phony
+  target](https://makefiletutorial.com/#phony), meaning `serve` is not a file we
+  want to generate but just a label for an action. In a simple `Makefile` like
+  this one, declaring phony targets isn't strictly necessary.
+- The line under `serve:` is the *recipe*, basically the command that will be
+  executed when we run `make serve` at the command line.
+  - `app=counter` sets the `app` environment variable to `counter`,
+    which is the name of the directory this `Makefile` lives in
+  - The [-C
+    option](https://www.gnu.org/software/make/manual/html_node/Options-Summary.html)
+    tells Make to change the directory before reading the `Makefile`. In effect,
+    it means we're running the `serve` rule in the root directory's `Makefile`.
+
+## Root directory `Makefile`
+
+Currently, the `serve` rule in root directory's `Makefile` can only run the app
+in the `src` directory. We want it to be able to run the app in `src/counter`,
+so we need to change it:
+
+```make
+.PHONY: serve
+serve: ## Serve the application with a local HTTP server
+	npx webpack serve --open --mode development --entry ./_build/default/src/$(app)/output/src/$(app)/Index.js
+```
+
+We basically copied the `serve` npm command from `package.json` to create the
+recipe for this rule, but instead of a fixed value for the `--entry` option, we
+use
+
+```
+./_build/default/src/$(app)/output/src/$(app)/Index.js
+```
+
+This means that the entry script served by webpack dev server depends on the
+`app` environment variable, which is provided by `src/counter/Makefile`.
+
+You're now ready to run the new Counter app you created! Go into the
+`src/counter` directory and run
+
+```
+make serve
+```
+
+It should open a new tab in your default browser to <a
+href="http://localhost:8080/" target="_blank" rel="noreferrer
+noopener">http://localhost:8080/</a> and display your Counter component.
+
+## Serving a default app
+
+If you run `make serve` in the root directory, you'll get an error that looks
+something like this:
+
+```
+ERROR in main
+Module not found:
+Error: Can't resolve './_build/default/src//output/src//Index.js' in '...'
+```
+
+That's because if `app` environment variable isn't provided, the value passed to
+the `--entry` option is no longer a valid path. We can define a default value
+for `app` by putting this near the top of `Makefile`:
+
+```make
+ifndef app
+	app := counter
+endif
+```
```

Comment:
is this necessary? Seems very long and not adding much to the setup

Reply:
Do you mean the "Serving a default app" section? 

I agree it's not essential, and could be deleted without losing much.

Reply:
Deleted 'Serving a default app' section in https://github.com/melange-re/melange-for-react-devs/pull/6/commits/00be2116875700b99caa20daefcc2a813ef70892